tidb-server/main.go

```diff
@@ -141,11 +145,21 @@ func main() {
 	}
 	registerStores()
 	registerMetrics()
-	loadConfig()
+	configWarning := loadConfig()
 	overrideConfig()
 	validateConfig()
+	if *configCheck {
```

Comment:
What if `configCheck` is true, and the `configStrict` is false?

Reply:
So we need to use `tidb-server --configCheck true --configStrict true` to verify the config file?

Reply:
No. I can't think of any reason to use --config-check=true and --config-strict=true at the same time.

--config-check tells the server to exit after validating the configuration file. It should exit with a 0 exit code if the config file is successfully validated, and it should exit with a non-zero exit code if the config file is not successfully validated.

--config-strict tells the server that it should *refuse* to start if the config file cannot be validated.

Reply:
Thanks or bringing this up, I added a line to basically ignore --config-strict if --config-check is true:

`                if _, ok := err.(*config.ErrConfigValidationFailed); ok && !*configCheck && !*configStrict {`